src/overlays/actors/ovl_player_actor/z_player.c

```diff
@@ -3371,23 +3371,23 @@ s32 Player_SetupAction(PlayState* play, Player* this, PlayerActionFunc actionFun
     return 1;
 }
 
-void func_80835DAC(PlayState* play, Player* this, PlayerActionFunc actionFunc, s32 flags) {
-    s32 temp;
+void Player_SetupActionPreserveAnimMovement(PlayState* play, Player* this, PlayerActionFunc actionFunc, s32 flags) {
+    s32 savedAnimMovementFlags;
 
-    temp = this->skelAnime.movementFlags;
+    savedAnimMovementFlags = this->skelAnime.movementFlags;
     this->skelAnime.movementFlags = 0;
     Player_SetupAction(play, this, actionFunc, flags);
-    this->skelAnime.movementFlags = temp;
+    this->skelAnime.movementFlags = savedAnimMovementFlags;
 }
 
-void func_80835DE4(PlayState* play, Player* this, PlayerActionFunc actionFunc, s32 flags) {
-    s32 temp;
+void Player_SetupActionPreserveItemAction(PlayState* play, Player* this, PlayerActionFunc actionFunc, s32 flags) {
+    s32 savedItemAction;
 
     if (this->itemAction >= 0) {
-        temp = this->itemAction;
+        savedItemAction = this->itemAction;
         this->itemAction = this->heldItemAction;
         Player_SetupAction(play, this, actionFunc, flags);
-        this->itemAction = temp;
+        this->itemAction = savedItemAction;
         Player_SetModels(this, Player_ActionToModelGroup(this, this->itemAction));
     }
 }
```

Comment:
the talking changes seem good to me but I'm not sure what these `SetupActionPreserveX` functions are for based on the name. And the implementation seems to override the anim flags and item action in addition to restoring them? Could you explain how they're used?

Reply:
It saves those things before changing action (which would reset them), and then restores them when the action change is done, to preserve its state from before the action change.
Not sure what you mean by an override is occurring, can you explain that

Reply:
I mean, why does it also set `this->skelAnime.movementFlags = 0` or `this->itemAction = this->heldItemAction` or before calling `Player_SetupAction`? If the only goal is to restore the state it seems like that wouldn't be necessary, so is there more going on with that?

Reply:
Oh I see.
Yeah for anim movement, its trying to skip `Player_FinishAnimMovement` because that checks if `animMovementFlags` are 0. 
For item actions, its trying to avoid `func_8008EC70` from being called, which is blocked by a `this->itemAction != this->heldItemAction` check.
Essentially just temporarily setting them to avoid resetting routines from happening.

I can separate this out to a new PR if its too much to bundle with the talking things.

Reply:
Maybe add those things as a comment? Or split it out into a separate PR, up to you (since the rest looks good to me here)

Reply:
yeah will do in a new PR